src/SignalR/server/Core/src/HubHandle.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    /// <summary>
+    /// A handle to the hub instance.
+    /// </summary>
+    /// <typeparam name="THub"></typeparam>
+    public readonly struct HubHandle<THub> where THub : Hub
```

Comment:
Any thought about making this a class so an alternative IHubActivator implementation could add it's own activation-related metadata to the retrieved hub? Or would that be too slow? We could make Created internal if we made HubHandle a class.

Reply:
Allocations bro. I don't want to complicated pooling to avoid allocations. Being a struct solves both problems... I did consider making this `HubHandle<THub, TState>`.

Reply:
I understand the allocation concern. Just wanted to make sure this was considered.

`HubHandle<THub, TState>` is interestring, but TState would have to be known either when retrieving the IHubActivator from DI (if we added the TState to the type) or when calling IHubActivator.Create() (if we added TState the just the method signature).

Either way, as an IHubActivator implementer, it feels like you're locked into using whatever TState DefaultHubActivator uses, or else you're going to break all the calling code.

Reply:
When people ask for the feature to extend the struct, we can offer a state object.